### ISSUE TITLE
IO current time support

### ISSUE THREAD

**Issue by Anderssorby** [leanprover/lean4#2123]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Checked that your issue isn't already [filed](https://github.com/leanprover/lean4/issues).
    * Reduced the issue to a self-contained, reproducible test case.

### Description

Add a `@[extern "lean_current_time_millis"] opaque IO.currentTimeMillis : IO UInt64` to read the system clock. This is useful for application development, logging and generating randomness.

```c
#include <time.h>
extern lean_obj_res lean_current_time_millis() {
    time_t t;
    time(&t);
    return lean_io_mk_ok(lean_box((uint64_t) t));
}

```

**Comment by Vtec234** [comment]
Do you need that time to be related to "real" time somehow? If not there is [`monoMsNow`](https://github.com/leanprover/lean4/blob/473486eeb9bca08d8557a3cd470c2512557c98ca/src/Init/System/IO.lean#L169).

**Comment by Anderssorby** [comment]
Yes, it is intended to give an actual timestamp. I want to use it for more generic application development.

**Comment by Kha** [comment]
#4904
